components/chassis.py

```diff
@@ -276,11 +275,27 @@ def execute(self) -> None:
         else:
             desired_speeds = self.chassis_speeds
 
-        desired_states = self.kinematics.toSwerveModuleStates(desired_speeds)
-        desired_states = self.kinematics.desaturateWheelSpeeds(
-            desired_states, attainableMaxSpeed=self.max_wheel_speed
-        )
+        if self.swerve_lock:
+            # Actuating the swerve to face to the center of the robot, not movable
+            desired_states = (
+                SwerveModuleState(0, Rotation2d(self.modules[0].central_angle)),
+                SwerveModuleState(0, Rotation2d(self.modules[1].central_angle)),
+                SwerveModuleState(0, Rotation2d(self.modules[2].central_angle)),
+                SwerveModuleState(0, Rotation2d(self.modules[3].central_angle)),
+            )
+
+            self.do_smooth = False
+        else:
+            desired_states = self.kinematics.toSwerveModuleStates(desired_speeds)
+            desired_states = self.kinematics.desaturateWheelSpeeds(
+                desired_states, attainableMaxSpeed=self.max_wheel_speed
+            )
+
         for state, module in zip(desired_states, self.modules):
+            if abs(state.speed) < 0.01 and not self.swerve_lock:
+                module.drive.set(ctre.ControlMode.Velocity, 0)
+                module.steer.set(ctre.ControlMode.PercentOutput, 0)
```

Comment:
This logic really doesn't belong here. The modules should know how to handle this themselves.

Can we add a lock method to the modules instead?